lib/main.go

```diff
@@ -200,6 +200,41 @@ func (obj *Main) Exit(err error) {
 	obj.exit <- err // trigger an exit!
 }
 
+// logic to create (tmp) prefix directory
+func (obj *Main) createPrefix(hostname string) (string, error) {
+	var prefix string
```

Comment:
Depending on what the end function looks like, in general it is preferable to avoid declaring variables ahead of time, and instead rely on `:=` https://tour.golang.org/basics/10

If it gets reused in a different scope (eg: an if branch), it will be fine due to shadowing.